undertow/src/main/java/org/wildfly/extension/undertow/filters/ModClusterNodeDefinition.java

```diff
@@ -49,306 +65,251 @@
             @Override
-            protected void handleNode(OperationContext context, ModClusterStatus.Node ctx, ModelNode operation) throws OperationFailedException {
-                context.getResult().set(new ModelNode(ctx.getRead()));
+            public ModelNode execute(ModClusterStatus.Node node) {
+                return new ModelNode(node.getTransferred());
             }
-        });
-        resourceRegistration.registerReadOnlyAttribute(REQUEST_QUEUE_SIZE, new AbstractNodeOperation() {
-
+        },
+        TTL(Constants.TTL, ModelType.LONG) {
             @Override
-            protected void handleNode(OperationContext context, ModClusterStatus.Node ctx, ModelNode operation) throws OperationFailedException {
-                context.getResult().set(new ModelNode(ctx.getRequestQueueSize()));
+            public ModelNode execute(ModClusterStatus.Node node) {
+                return new ModelNode(node.getTtl());
             }
-        });
-        resourceRegistration.registerReadOnlyAttribute(TIMEOUT, new AbstractNodeOperation() {
-
+        },
+        FLUSH_PACKETS(Constants.FLUSH_PACKETS, ModelType.BOOLEAN) {
             @Override
-            protected void handleNode(OperationContext context, ModClusterStatus.Node ctx, ModelNode operation) throws OperationFailedException {
-                context.getResult().set(new ModelNode(ctx.getTimeout()));
+            public ModelNode execute(ModClusterStatus.Node node) {
+                return node.isFlushPackets() ? ModelNode.TRUE : ModelNode.FALSE;
             }
-        });
-        resourceRegistration.registerReadOnlyAttribute(WRITTEN, new AbstractNodeOperation() {
-
+        },
+        QUEUE_NEW_REQUESTS(Constants.QUEUE_NEW_REQUESTS, ModelType.BOOLEAN) {
             @Override
-            protected void handleNode(OperationContext context, ModClusterStatus.Node ctx, ModelNode operation) throws OperationFailedException {
-                context.getResult().set(new ModelNode(ctx.getTransferred()));
+            public ModelNode execute(ModClusterStatus.Node node) {
+                return node.isQueueNewRequests() ? ModelNode.TRUE : ModelNode.FALSE;
             }
-        });
-        resourceRegistration.registerReadOnlyAttribute(TTL, new AbstractNodeOperation() {
-
+        },
+        URI(Constants.URI, ModelType.STRING) {
             @Override
-            protected void handleNode(OperationContext context, ModClusterStatus.Node ctx, ModelNode operation) throws OperationFailedException {
-                context.getResult().set(new ModelNode(ctx.getTtl()));
+            public ModelNode execute(ModClusterStatus.Node node) {
+                return new ModelNode(node.getUri().toString());
             }
-        });
-        resourceRegistration.registerReadOnlyAttribute(FLUSH_PACKETS, new AbstractNodeOperation() {
-
+        },
+        ALIASES(Constants.ALIASES, ModelType.LIST, PrimitiveListAttributeDefinition.Builder::new) {
```

Comment:
This inadvertently results in model type LIST of value type LIST instead of intended STRING.

Reply:
Fixed.